Integrations/Slack/Slack.py

```diff
@@ -187,6 +205,87 @@ def set_to_latest_integration_context(key: str, value, wait: bool = False):
     demisto.setIntegrationContext(integration_context)
 
 
+def set_name_and_icon(body, method):
+    """
+    If provided, sets a name and an icon for the bot if a message is sent.
+    :param body: The message body.
+    :param method: The current API method.
+    """
+    if method == 'chat.postMessage':
+        if BOT_NAME:
+            body['username'] = BOT_NAME
+        if BOT_ICON_URL:
+            body['icon_url'] = BOT_ICON_URL
+
+
+def send_slack_request_sync(client: slack.WebClient, method: str, http_verb: str = 'POST', file_: dict = None,
+                            body: dict = None) -> SlackResponse:
+    """
+    Sends a request to slack API while handling rate limit errors.
+    :param client: The slack client.
+    :param method: The method to use.
+    :param http_verb: The HTTP method to use.
+    :param file_: A file to send.
+    :param body: The request body.
+    :return: The slack API response.
+    """
+    set_name_and_icon(body, method)
+    while True:
+        try:
+            if http_verb == 'POST':
+                if file_:
+                    response = client.api_call(method, files={"file": file_}, data=body)
+                else:
+                    response = client.api_call(method, json=body)
+            else:
+                response = client.api_call(method, http_verb='GET', params=body)
+        except SlackApiError as api_error:
+            response = api_error.response
+            if 'Retry-After' in response.headers:
+                retry_after = int(response.headers['Retry-After'])
+                if retry_after < MAX_LIMIT_TIME:
```

Comment:
Logic seems flawed here. You should have a counter: `total_retry_time` which increases with each try. So if we did a request and got back retry after 20 seconds and then we do another request and get back retry after 45 seconds  we won't try the second request if max limit is 60.